### PR TITLE
[65713] Add Event conferencing support

### PR DESCRIPTION
**Pull request by mrashed-dev**
# Description
This PR enables support for the new Nylas [Conference Sync (Beta)](https://developer.nylas.com/docs/connectivity/calendar/conference-sync-beta) feature.

# Usage
`Event.conference` is now a field that will contain conference details for Events that contain them, and it can be populated when new events through the SDK.

You can read the values like so:
```java
public class NylasExamples {
  public static void getEventExample() throws IOException, RequestFailedException {
    NylasClient nylas = new NylasClient();
    NylasAccount account = nylas.account("{ACCESS_TOKEN}");
    // Get an event object with conferencing details
    Event event = account.events().get("{EVENT_ID}");
    // Parse the conferencing data
    Event.Conferencing conferencing = event.getConferencing();
    String conferencingProvider = conferencing.getProvider();
    Event.Conferencing.Details details = conferencing.details();
    String meetingCode = details.getMeetingCode();
    String password = details.getPassword();
    String pin = details.getPin();
    String url = details.getUrl();
    List<String> phone = details.getPhone();
  }
}
```
You can also build a new event that includes conferencing details with the SDK:
```java
public class NylasExamples {
  public static void postEventExample() throws IOException, RequestFailedException {
    NylasClient nylas = new NylasClient();
    NylasAccount account = nylas.account("{ACCESS_TOKEN}");
    // Create a new event object
    Event event = new Event("{CALENDAR_ID}", when);
    // Add conferencing details
    Event.Conferencing conferencing = new Event.Conferencing();
    conferencing.setProvider("Zoom Meeting");

    Event.Conferencing.Details details = new Event.Conferencing.Details();
    details.setMeetingCode("213");
    details.setPassword("xyz");
    details.setPin("555");
    details.setUrl("https://us02web.zoom.us/j/****************");
    details.setPhone(Collections.singletonList("+11234567890"));
    conferencing.setDetails(details);

    event.setConferencing(conferencing);
  }
}
```

# License
I confirm that this contribution is made under the terms of the MIT license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: src/main/java/com/nylas/EventConferencing.java [hunk 1]
```diff
@@ -0,0 +1,86 @@
+package com.nylas;
+
+import java.util.List;
+import java.util.Map;
+
+public class EventConferencing {
```
Comment (ddlatham) [inline src/main/java/com/nylas/EventConferencing.java:6]: I suggest moving this to a nested class in Events called Conferencing similar to other Event subobjects

File: src/examples/java/com/nylas/examples/other/EventsExample.java [hunk 2]
```diff
@@ -102,6 +93,16 @@ protected static void basicEventCrud(Events events, Calendar primary) throws IOE
...
 
+		EventConferencing conferencing = new EventConferencing();
+		conferencing.setProvider("Zoom Meeting");
+		EventConferencing.Details details = new EventConferencing.Details();
+		details.setMeetingCode("213");
+		details.setPassword("xyz");
+		details.setUrl("https://us02web.zoom.us/j/****************");
+		details.setPhone(Collections.singletonList("+11234567890"));
+		conferencing.setDetails(details);
+		event.setConferencing(conferencing);
```
Comment (ddlatham) [inline src/examples/java/com/nylas/examples/other/EventsExample.java:104]: this should be on the `created` variable, rather than the `event` object.

**Review by ddlatham — CHANGES_REQUESTED** [review]
Just a couple comments.  Overall looks good.

File: src/examples/java/com/nylas/examples/other/EventsExample.java (hunk 2 above)
Comment (mrashed-dev) [reply src/examples/java/com/nylas/examples/other/EventsExample.java:104]: Good catch, copy/paste error on my part lol

**Review by ddlatham — APPROVED** [review]
Looks good to me.  There is one typo in a comment "deserialiation" if you want to fix.

**Pull request merged**